exchange/exchange.go

```diff
@@ -903,6 +903,66 @@ func (e *exchange) getBidCacheInfo(bid *pbsOrtbBid, auction *auction) (cacheInfo
 	return
 }
 
+// getAuctionCurrencyRates returns the constant currency conversion rates that will be used to process this
+// auction. Parameter customRates contains, if any, the bidRequest custom currency conversion rates
+// defined in bidRequest.Ext.  If usePbsRates flag is set to false, all previous Conversions values are discarded
+// and this function merely substitutes Conversions with customRates. But if usePbsRates was ommitted or set to
+// true, customRates' values update or add to the Conversions map
+func (e *exchange) getAuctionCurrencyRates(customRates *openrtb_ext.ExtRequestCurrency) currency.Conversions {
+	// No bidRequest.ext-defined, use PBS rates
+	if customRates == nil || len(customRates.ConversionRates) == 0 {
+		return e.currencyConverter.Rates()
+	}
+
+	var usePbsRates bool = true
+	if customRates.UsePBSRates != nil {
+		usePbsRates = *customRates.UsePBSRates
+	}
+
+	// when usePbsRates is false, only use bidRequest.ext-defined rates
+	if !usePbsRates {
+		return currency.NewRates(time.Now(), customRates.ConversionRates)
```

Comment:
Why is the time needed for constant rates from the request?

Reply:
Although these custom rates are constant values that will not change, in terms of the logic inside our `currency` package, they are not considered `ConstantRates` or at least they are not named that. This return value is meant to return the `currency.Conversions` implementation found in `currency/rates.go`. While "constant" rates are implemented in `currency/constant_rates.go`. They widely differ and in my opinion `ConstantRates` should be renamed to something else. Probably `SameCurrencyRates` or `UnitaryRates` or something. Kept that part out of the scope of this PR but I could always modify it if you disagree.

Because line 924 returns all-custom currency conversions object that doesn't go into a `RateConverter` dating them with `time.Now()` as the time they were `fetched` seemed more accurate but you raise a good point. Given that we don't actually do anything with that time stamp from here on I'll change to a less expensive `time.Time{}` instead.

Reply:
Ideally, I think we should remove the reference to time entirely.

It seems this field only exists to fully parse the auto generated PBS currency file. We don't actually do anything with that timestamp at all. Perhaps we can remove it entirely.

Thoughts? @guscarreon @bsardo

Reply:
I didn't remove it in today's PR update but I can do so if you want to. @bsardo @SyntaxNode please let me know.